### PR TITLE
Stylelint for missed MDS color usages

### PR DESCRIPTION
**Pull request by juanca**
https://www.pivotaltracker.com/story/show/172248579

## PR upkeep checklist

- [x] Change log entry
- [x] Label(s)
- [x] Assignee(s)
- [x] Deployment URL: https://mavenlink.github.io/design-system/webinf-peanut-butter-linter-with-a-baseball-bat-172248579
- [x] (When ready for review) Reviewer(s)

### CODE REVIEW

**Comment by juanca** [comment]
Map on converting colors v1 to v2 for greys: 

```
  --mds-grey-87: 123;
  --palette-grey-x-dark: rgba(0, 0, 0, 0.87);
  --palette-neutral-xx-dark: #282624;
  --palette-neutral-x-dark: #333;

  --mds-grey-54;
  --palette-grey-dark: rgba(0, 0, 0, 0.54);
  --palette-neutral-dark: #4a4a4a;
  --palette-neutral-mid-dark: #737373;

  --mds-grey-38;
  --palette-grey-base: rgba(0, 0, 0, 0.38);
  --palette-neutral-base: #9d9d9d;
  --palette-neutral-mid: #b6b6b6;

  --mds-grey-12;
  --palette-grey-light: rgba(0, 0, 0, 0.12);
  --palette-neutral-mid-light: #c8c8c8;
  --palette-neutral-light: #e1e1dd;

  --mds-grey-03;
  --palette-grey-x-light: rgba(0, 0, 0, 0.03);
  --palette-neutral-x-light: #f5f5f2;
  --palette-neutral-xx-light: #fbfaf6;
```

File: styleguide/content/overview/overview.css [hunk 1]
```diff
@@ -31,7 +31,7 @@
   display: block;
   width: 45px;
   margin-bottom: 20px;
-  border-bottom: 2px solid var(--palette-neutral-dark);
+  border-bottom: 2px solid var(--mds-grey-87);
```
Comment (juanca) [inline styleguide/content/overview/overview.css:34]: We want 54.

File: styleguide/components/components-list/components-list.css [hunk 2]
```diff
@@ -28,17 +28,17 @@
 
 .link:link,
 .link:visited {
-  color: var(--palette-neutral-mid-dark);
+  color: var(--mds-grey-38);
```
Comment (juanca) [inline styleguide/components/components-list/components-list.css:31]: We want 54.

File: styleguide/components/header/header.css [hunk 3]
```diff
@@ -28,21 +29,21 @@
 }
 
 .end {
-  color: var(--palette-neutral-mid-dark);
+  color: var(--mds-grey-38);
```
Comment (juanca) [inline styleguide/components/header/header.css:32]: 54.

Comment (juanca) [reply styleguide/components/header/header.css:32]:
```suggestion
  color: var(--mds-grey-54);
```

File: styleguide/components/components-list/components-list.css (hunk 2 above)
Comment (juanca) [reply styleguide/components/components-list/components-list.css:31]:
```suggestion
  color: var(--mds-grey-54);
```

File: styleguide/content/brand-identity/typography.css [hunk 4]
```diff
@@ -153,11 +154,11 @@
 
 :global(.type-headings) {
   margin-bottom: 0;
-  border-bottom: 1px solid var(--palette-neutral-mid-dark);
+  border-bottom: 1px solid var(--mds-grey-87);
```
Comment (juanca) [inline styleguide/content/brand-identity/typography.css:157]:
```suggestion
  border-bottom: 1px solid var(--mds-grey-54);
```

File: styleguide/components/header/header.css [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 
 :root {
   --headerHeight: 65px;
 }
 
 .header {
-  background-color: var(--palette-neutral-xx-light);
-  border-bottom: 1px solid var(--palette-neutral-light);
+  background-color: var(--mds-grey-3);
+  border-bottom: 1px solid var(--mds-grey-3);
```
Comment (juanca) [inline styleguide/components/header/header.css:11]:
```suggestion
  border-bottom: 1px solid var(--mds-grey-12);
```

File: styleguide/components/header/header.css [hunk 6]
```diff
@@ -28,21 +29,21 @@
...
+  color: var(--mds-grey-87);
   line-height: var(--headerHeight);
   padding: 0 0.5em;
   text-decoration: none;
   transition: background-color 150ms linear;
 }
 
 .github:hover {
-  background-color: var(--palette-neutral-light);
+  background-color: var(--mds-grey-8);
```
Comment (juanca) [inline styleguide/components/header/header.css:48]:
```suggestion
  background-color: var(--mds-grey-12);
```

File: styleguide/components/table/table.css [hunk 7]
```diff
@@ -1,8 +1,8 @@
-@import "../../../src/styles/colors.css";
+@import "../../../src/styles/colors-v2.css";
 
 .table {
   composes: table from '../../../src/components/table/table.css';
-  border: 1px solid var(--palette-neutral-light);
+  border: 1px solid var(--mds-grey-8);
```
Comment (juanca) [inline styleguide/components/table/table.css:5]:
```suggestion
  border: 1px solid var(--mds-grey-12);
```

File: styleguide/content/brand-identity/typography.css [hunk 8]
```diff
@@ -1,16 +1,17 @@
...
+@import '../../../src/styles/colors-v2.css';
 @import "../../../src/styles/typography.css";
 
 :global(.type-top) {
   margin-top: 60px;
 }
 
 :global(.type-title) {
-  border-bottom: 1px solid var(--palette-neutral-light);
+  border-bottom: 1px solid var(--mds-grey-8);
```
Comment (juanca) [inline styleguide/content/brand-identity/typography.css:10]:
```suggestion
  border-bottom: 1px solid var(--mds-grey-12);
```

File: styleguide/content/brand-identity/typography.css [hunk 9]
```diff
@@ -99,7 +100,7 @@
   margin-bottom: 0;
   height: 75px;
   align-items: center;
-  border-bottom: 1px solid var(--palette-neutral-light);
+  border-bottom: 1px solid var(--mds-grey-8);
```
Comment (juanca) [inline styleguide/content/brand-identity/typography.css:103]:
```suggestion
  border-bottom: 1px solid var(--mds-grey-12);
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -5,7 +5,9 @@
     Click to see more.
   </summary>
 
-  - This is a change log example item
+  - Minor: Create linter for new MDS colors
```
Comment (Bloodyaugust) [inline CHANGELOG.md:8]: I like this new approach! Calling out the severity of each change is a great idea.

File: package.json [hunk 11]
```diff
@@ -62,7 +62,7 @@
     "react-test-renderer": "16.8.0",
     "react-transition-group": "^2.5.3",
     "style-loader": "^0.23.1",
-    "stylelint": "^10.1.0",
+    "stylelint": "^13.3.3",
```
Comment (Bloodyaugust) [inline package.json:65]: What version of node is required to support this version of `stylelint`?

File: styleguide/components/components-list/components-list.css [hunk 12]
```diff
@@ -28,17 +28,17 @@
...
-  border-left: var(--selectedBorderWidth) solid var(--palette-primary-base);
+  border-left: var(--selectedBorderWidth) solid var(--mds-brand-100);
   font-weight: bold;
   padding-left: 3px;
 }
 
 .current:link,
 .current:visited {
-  color: black;
+  color: var(--black);
```
Comment (Bloodyaugust) [inline styleguide/components/components-list/components-list.css:43]: I still think it's kind of weird to define our own primitives variables over the exact same primitive values that are available as CSS named colors.

File: styleguide/content/brand-identity/typography.css [hunk 13]
```diff
@@ -143,7 +144,7 @@
 :global(.type-columns > span:nth-of-type(4)) {
   flex-basis: 25%;
   flex-grow: 1;
-  color: #716f67;
+  color: #716f67; /* stylelint-disable-line mds/colors */
```
Comment (Bloodyaugust) [inline styleguide/content/brand-identity/typography.css:147]: We use this color more than once, should we define a new color variable for it?

Comment (juanca) [reply styleguide/content/brand-identity/typography.css:147]: A couple of things about this particular instance:

1. It is for the typography page that is going to be ripped out once we get documentation from designers
2. We are skipping our rule on this color -- we didn't add the color
3. We actually have no idea where this color is being applied.

File: styleguide/components/components-list/components-list.css (hunk 12 above)
Comment (juanca) [reply styleguide/components/components-list/components-list.css:43]: I think it's a design choice: if we want to make black slight less black then we can do it through variables. I'm not sure how much value that actually gives in modern times but it's been a thing. Maybe that's feedback for the design team.

File: package.json (hunk 11 above)
Comment (juanca) [reply package.json:65]: There isn't a node version specified in their package.json

Is this a concern? It seems to be working for our dev environments.

Comment (Bloodyaugust) [reply package.json:65]: Just trying to keep aware. I think CI is ~node10. We should probably push to standardize node version in bigmaven. 😅

Comment (juanca) [reply package.json:65]: Yeah, I agree! 

We know if it runs in MDS then it is Node 10. If it runs in bigmaven then it is Node 8. 

Stylelint it pretty popular so I'm going to assume it is safe to run LTS version of Node for it -- and that is usually what _we_ should be running in production environments.

**Pull request merged**